contracts/src/v0.8/ccip/pools/BurnToAddressMintTokenPool.sol

```diff
@@ -0,0 +1,99 @@
+
+import {ITypeAndVersion} from "../../shared/interfaces/ITypeAndVersion.sol";
+import {IBurnMintERC20} from "../../shared/token/ERC20/IBurnMintERC20.sol";
+
+import {Pool} from "../libraries/Pool.sol";
+import {BurnMintTokenPoolAbstract} from "./BurnMintTokenPoolAbstract.sol";
+import {TokenPool} from "./TokenPool.sol";
+
+import {IERC20} from "../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @notice This pool mints and burns a 3rd-party token by sending tokens to an address which is unrecoverable.
+/// @dev The pool is designed to have an immutable burn address. If the tokens at the burn address become recoverable,
+/// for example, a quantum computer calculating a private key for the zero address, the pool will need to be replaced
+/// with a new pool with a different burn address.
+contract BurnToAddressMintTokenPool is BurnMintTokenPoolAbstract, ITypeAndVersion {
+  using SafeERC20 for IERC20;
+
+  event MintedTokensSet(uint256 newMintedTokenAmount, uint256 oldMintedTokenAmount);
+
+  error InsufficientMintedTokens();
+
+  string public constant override typeAndVersion = "BurnToAddressTokenPool 1.5.1";
+
+  /// @notice The address where tokens are sent during a call to lockOrBurn, functionally burning but without decreasing
+  /// total supply. This address is expected to have no ability to recover the tokens sent to it, and will thus be locked forever.
+  /// This can be either an EOA without a corresponding private key, or a contract which does not have the ability to transfer the tokens.
+  address public immutable i_burnAddress;
+
+  /// @notice Minted Tokens is a safety mechanism to ensure that more tokens cannot be sent out of the bridge
+  /// than were originally sent in via CCIP. On incoming messages the value is increased, and on outgoing messages,
+  /// the value is decreased. For pools with existing tokens in circulation, the value may not be known at deployment
+  /// time, and thus should be set later using the setMintedTokens() function.
+  uint256 internal s_mintedTokens;
+
+  /// @dev Since burnAddress is expected to make the tokens unrecoverable, no check for the zero address needs to be
+  /// performed, as it is a valid input.
+  constructor(
+    IBurnMintERC20 token,
+    uint8 localTokenDecimals,
+    address[] memory allowlist,
+    address rmnProxy,
+    address router,
+    address burnAddress
+  ) TokenPool(token, localTokenDecimals, allowlist, rmnProxy, router) {
+    i_burnAddress = burnAddress;
+  }
+
+  /// @notice Mint tokens from the pool to the recipient, updating the internal accounting for an outflow of tokens.
+  /// @dev If the amount of tokens to be
+  function releaseOrMint(
+    Pool.ReleaseOrMintInV1 calldata releaseOrMintIn
+  ) public virtual override returns (Pool.ReleaseOrMintOutV1 memory) {
+    s_mintedTokens += releaseOrMintIn.amount;
```

Comment:
Let's add comments to the += here and the -= below because this took me some effort to wrap my head around. Maybe changing the name helps but something along the lines of 
// minting tokens on this chain increased the outstanding supply of tokens

and 

// when tokens are sent to another chain, it decreases the local token supply